sycl/unittests/scheduler/InOrderQueueSyncCheck.cpp

```diff
@@ -61,12 +68,16 @@ TEST_F(SchedulerTest, InOrderQueueSyncCheck) {
   sycl::event Event;
   {
     LimitedHandlerSimulation MockCGH;
-    EXPECT_CALL(MockCGH, depends_on).Times(0);
-    Queue->finalizeHandler<LimitedHandlerSimulation>(MockCGH, Event);
+    EXPECT_CALL(MockCGH, depends_on(An<const sycl::detail::EventImplPtr &>()))
+        .Times(0);
+    Queue->finalizeHandler<LimitedHandlerSimulation>(
+        MockCGH, detail::CG::CGTYPE::CodeplayHostTask, Event);
   }
   {
     LimitedHandlerSimulation MockCGH;
-    EXPECT_CALL(MockCGH, depends_on).Times(1);
-    Queue->finalizeHandler<LimitedHandlerSimulation>(MockCGH, Event);
+    EXPECT_CALL(MockCGH, depends_on(An<const sycl::detail::EventImplPtr &>()))
+        .Times(1);
+    Queue->finalizeHandler<LimitedHandlerSimulation>(
+        MockCGH, detail::CG::CGTYPE::CodeplayHostTask, Event);

```

Comment:
Do we test scenarios with multiple in-order queues, where host task enqueued in one queue and event was passed as an argument to a barrier in another queue?